scenario-compiler/src/test/scala/pl/touk/nussknacker/engine/InterpreterSpec.scala

```diff
@@ -1020,6 +1024,100 @@ class InterpreterSpec extends AnyFunSuite with Matchers {
     interpretProcess(process, Transaction()) shouldBe "someKey"
   }
 
+  test("spel template ast operation parameter should work for template and literal value") {
+    val process = ScenarioBuilder
+      .streaming("test")
+      .source("start", "transaction-source")
+      .enricher(
+        "ex",
+        "out",
+        "templateAstOperationService",
+        "template" -> Expression.spelTemplate(s"Hello#{#input.msisdn}")
+      )
+      .buildSimpleVariable("result-end", resultVariable, "#out".spel)
+      .emptySink("end-end", "dummySink")
+
+    interpretProcess(process, Transaction(msisdn = "foo")) should equal("[Hello]-literal[foo]-templated")
+  }
+
+  test("spel template AST operation parameter should handle multiple cases") {
+    val testCases = Seq(
+      (
+        "templated value and literal value",
+        s"Hello#{#input.msisdn}",
+        Transaction(msisdn = "foo"),
+        "[Hello]-literal[foo]-templated"
+      ),
+      (
+        "single literal value",
+        "Hello",
+        Transaction(msisdn = "foo"),
+        "[Hello]-literal"
+      ),
+      (
+        "single templated function call expression",
+        "#{#input.msisdn.toString()}",
+        Transaction(msisdn = "foo"),
+        "[foo]-templated"
+      ),
+      (
+        "empty value",
+        "",
+        Transaction(msisdn = "foo"),
+        "[]-literal"
+      ),
+    )
+    for ((description, templateExpression, inputTransaction, expectedOutput) <- testCases) {
+      withClue(s"Test case: $description") {
+        val process = ScenarioBuilder
+          .streaming("test")
+          .source("start", "transaction-source")
+          .enricher(
+            "ex",
+            "out",
+            "templateAstOperationService",
+            "template" -> Expression.spelTemplate(templateExpression)
+          )
+          .buildSimpleVariable("result-end", resultVariable, "#out".spel)
+          .emptySink("end-end", "dummySink")
+
+        interpretProcess(process, inputTransaction) should equal(expectedOutput)
+      }
+    }
+  }
+
+  test("spel template ast operation parameter should work for single literal value") {
+    val process = ScenarioBuilder
+      .streaming("test")
+      .source("start", "transaction-source")
+      .enricher(
+        "ex",
+        "out",
+        "templateAstOperationService",
+        "template" -> Expression.spelTemplate("Hello")
+      )
+      .buildSimpleVariable("result-end", resultVariable, "#out".spel)
+      .emptySink("end-end", "dummySink")
+
+    interpretProcess(process, Transaction(msisdn = "foo")) should equal("[Hello]-literal")
+  }
+
+  test("spel template ast operation parameter should work for single templated function call expression") {
```

Comment:
It would be nice to have a test for the case when `PostponedEvaluatorLazyParameterStrategy` is used. AFAIK it is used in source/sink/custom nodes combined with Flink engine.

Reply:
Yes, this test is in TemplateLazyParameterTest. But weirdly, the test passes without changes in EvaluableLazyParameterCreator :/ still trying to figure it out

Also I parameterized all test cases into one test and forgot to delete these tests here

Reply:
Way more complex than I hoped :( need some advice. let's talk about it tomorrow if you'll have the time

Reply:
Done: https://github.com/TouK/nussknacker/pull/7174/files